torchao/prototype/mx_formats/constants.py

```diff
@@ -0,0 +1,48 @@
+import torch
+
+# This is conceptually an enum of non-core dtypes
+# if someone has time to verify torch.compile compatibility, it could be made
```

Comment:
The weird thing is that some of these dtypes are in core (the float8 ones) and some aren't (float6/float4/mx's spec of int8, etc).  I think it would be nice to have a clean structure unifying all of that, I just haven't had the time.  Definitely open for someone (or future me) to improve this.